x-pack/plugins/apm/public/components/app/settings/storage_explorer/services_table/index.tsx

```diff
@@ -0,0 +1,295 @@
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const comparisonEnabled = getComparisonEnabled({ core });
+
+  const toggleRowDetails = (selectedServiceName: string) => {
+    const expandedRowMapValues = { ...itemIdToExpandedRowMap };
+    if (expandedRowMapValues[selectedServiceName]) {
+      delete expandedRowMapValues[selectedServiceName];
+    } else {
+      expandedRowMapValues[selectedServiceName] = (
+        <StorageDetailsPerService
+          serviceName={selectedServiceName}
+          indexLifecyclePhase={indexLifecyclePhase}
+        />
+      );
+    }
+    setItemIdToExpandedRowMap(expandedRowMapValues);
+  };
+
+  const { data, status } = useProgressiveFetcher(
+    (callApmApi) => {
+      return callApmApi('GET /internal/apm/storage_explorer', {
+        params: {
+          query: {
+            indexLifecyclePhase,
+            start,
+            end,
+            environment,
+            kuery,
+          },
+        },
+      });
+    },
+    [indexLifecyclePhase, start, end, environment, kuery]
+  );
+
+  useEffect(() => {
+    // Closes any open rows when fetching new items
+    setItemIdToExpandedRowMap({});
+  }, [status]);
+
+  const loading =
+    status === FETCH_STATUS.NOT_INITIATED || status === FETCH_STATUS.LOADING;
+
+  const columns: Array<EuiBasicTableColumn<ValuesType<StorageExplorerItems>>> =
+    [
+      {
+        field: 'serviceName',
+        name: i18n.translate(
+          'xpack.apm.settings.storageExplorer.table.serviceColumnName',
+          {
+            defaultMessage: 'Service',
+          }
+        ),
+        sortable: true,
+        render: (_, { serviceName, agentName }) => {
+          const serviceLinkQuery = {
+            comparisonEnabled,
+            environment,
+            kuery,
+            rangeFrom,
+            rangeTo,
+            serviceGroup: '',
+          };
+
+          return (
+            <TruncateWithTooltip
+              data-test-subj="apmStorageExplorerServiceLink"
+              text={serviceName || NOT_AVAILABLE_LABEL}
+              content={
+                <ServiceLink
+                  query={serviceLinkQuery}
+                  serviceName={serviceName}
+                  agentName={agentName}
+                />
+              }
+            />
+          );
+        },
+      },
+      {
+        field: 'environment',
+        name: i18n.translate(
+          'xpack.apm.settings.storageExplorer.table.environmentColumnName',
+          {
+            defaultMessage: 'Environment',
+          }
+        ),
+        render: (_, { environments }) => (
+          <EnvironmentBadge environments={environments ?? []} />
+        ),
+        sortable: true,
+      },
+
+      {
+        field: 'sampling',
+        name: (
+          <EuiToolTip
+            content={i18n.translate(
+              'xpack.apm.settings.storageExplorer.table.samplingColumnDescription',
+              {
+                defaultMessage:
```

Comment:
Could use some help here @bmorelli25

Reply:
Is this showing the sample rate of the service/env? I think it might make sense to change the table column name from `Sampling` to `Sample rate`. Text could be something like, "The transaction sample rate". If we can add a link, I'd add a link to https://www.elastic.co/guide/en/apm/guide/current/configure-tail-based-sampling.html. "The transaction sample rate. To learn how to adjust this rate, see [Transaction sampling]."

Reply:
@bmorelli25 thanks, instead of displaying the sample rate configured for the service, we display the "real sampling" rate (total sampled transaction documents divided by the total throughput).

Reply:
Oh I see. No, I don't think so. It might be confusing to use the same terminology if this is calculated differently. Should we explain why this number can differ from the transaction sample rate? Something like this, perhaps:

> The number of sampled transactions divided by total throughput. This value may differ from the configured transaction sample rate because...

Reply:
Looking into the [docs](https://www.elastic.co/guide/en/apm/guide/current/sampling.html), we could use something like "The number of sampled transactions divided by total throughput. This value may differ from the configured transaction sample rate because it might be affected by the initial service's decision when using head-based sampling  or by a set of policies when using tail-based sampling."